rust/0226-invert-binary-tree.rs

```diff
@@ -0,0 +1,20 @@
+use std::rc::Rc;
+use std::cell::RefCell;
+impl Solution {
+    pub fn invert_tree(root: Option<Rc<RefCell<TreeNode>>>) -> Option<Rc<RefCell<TreeNode>>> {
+        if root == None {
```

Comment:
Instead of cloning the `root` option into `roots`, we can directly use pattern matching to check if the `root` option is `None` or `Some`. Then, we can use the `as_mut()` method on the Option to access the `RefCell`, and finally use the `borrow_mut()` method to obtain a mutable reference to the `TreeNode`. This way, we can avoid unnecessary cloning and unwrapping afterward.

This would be one way of doing this:
```rust
impl Solution {
    pub fn invert_tree(root: Option<Rc<RefCell<TreeNode>>>) -> Option<Rc<RefCell<TreeNode>>> {
        if let Some(mut root_ref) = root {
            let left = root_ref.borrow_mut().left.take();
            let right = root_ref.borrow_mut().right.take();

            root_ref.borrow_mut().left = Solution::invert_tree(right);
            root_ref.borrow_mut().right = Solution::invert_tree(left);

            Some(root_ref)
        } else {
            None
        }
    }
}

```